File: Wells/Source/VievModels/IEntitiesViewModel.vb
Public Interface IEntitiesViewModel

    ReadOnly Property IsNewCommandEnabled As Boolean
    ReadOnly Property IsRemoveCommandEnabled As Boolean

    ReadOnly Property NewEntityCommand As ICommand
    ReadOnly Property ImportEntitiesCommand As ICommand
    ReadOnly Property EditEntityCommand As ICommand
    ReadOnly Property RemoveEntityCommand As ICommand
End Interface

File: Wells/src/ViewModels/Entities/IEntitiesViewModel.cs
using System.Collections.Generic;
using System.Windows.Controls;
using System.Windows.Input;

namespace Wells.View.ViewModels
{
    public interface IEntitiesViewModel
    {
        bool IsNewCommandEnabled { get; }
        bool IsEditCommandEnabled { get; }
        ICommand NewEntityCommand { get; }
        ICommand EditEntityCommand { get; }
        ICommand RemoveEntityCommand { get; }
        ICommand ImportEntitiesCommand { get; }
        ICommand ExportEntitiesCommand { get; }
        ContextMenu GetContextMenu();

        void SetSelectedItems(IEnumerable<object> entities);
        void RemoveEventHandlers();
    }
}
